src/ru/desu/src/eu/kanade/tachiyomi/extension/ru/desu/Desu.kt

```diff
@@ -375,21 +374,19 @@ class Desu : ConfigurableSource, HttpSource() {
                 Toast.makeText(screen.context, warning, Toast.LENGTH_LONG).show()
                 true
             }
-        }
-        val domainDesuPref = EditTextPreference(screen.context).apply {
-            key = DOMAIN_TITLE
+        }.let(screen::addPreference)
+        EditTextPreference(screen.context).apply {
+            key = DOMAIN_PREF
             title = DOMAIN_TITLE
-            summary = domain
+            summary = baseUrl
             setDefaultValue(DOMAIN_DEFAULT)
             dialogTitle = DOMAIN_TITLE
-            setOnPreferenceChangeListener { _, _ ->
+            setOnPreferenceChangeListener { _, newValue ->
                 val warning = "Для смены домена необходимо перезапустить приложение с полной остановкой."
```

Comment:
Reject the change if the url is invalid
```suggestion
                if (!newValue.toString().startsWith("https://")) {
                    return@setOnPreferenceChangeListener false
                }
                val warning = "Для смены домена необходимо перезапустить приложение с полной остановкой."
```